with-snowflake/README.md

```diff
@@ -0,0 +1,315 @@
+```graphql
+query
+{
+  employee(EMAIL:"wsizeyf@sf_tuts.com")
+  {
+    FIRST_NAME
+    LAST_NAME
+    CITY
+  }
+} 
+```
+(with `stepzen request` we have compressed operations to simplify copying the command)
+
+The response should be:
+```json
+{
+  "data": {
+    "employee": {
+      "FIRST_NAME": "Wallis",
+      "LAST_NAME": "Sizey",
+      "CITY": "Taibao"
+    }
+  }
+}
+```
+
+We can page through all employees using `Query.employees` and standard GraphQL pagination.
+With StepZen using standard pagination allows clients to handle paging through results
+consistently regardless of if the backend is a database, REST api or another GraphQL API.
+
+First we show how to page through all employees, starting with the first five:
+```
+stepzen request 'query{employees(first:5){edges{node{FIRST_NAME LAST_NAME CITY}}pageInfo{endCursor hasNextPage}}}'
+```
+
+This executes:
+```graphql
+query {
+  employees(first: 5) {
+    edges {
+      node {
+        FIRST_NAME
+        LAST_NAME
+        CITY
+      }
+    }
+    pageInfo {
+      endCursor
+      hasNextPage
+    }
+  }
+}
+```
+
+The response should be similar to:
+```json
+{
+  "data": {
+    "employees": {
+      "edges": [
+        {
+          "node": {
+            "FIRST_NAME": "Hollis",
+            "LAST_NAME": "Anneslie",
+            "CITY": "Aleysk"
+          }
+        },
+        {
+          "node": {
+            "FIRST_NAME": "Di",
+            "LAST_NAME": "McGowran",
+            "CITY": "Banjar Bengkelgede"
+          }
+        },
+        {
+          "node": {
+            "FIRST_NAME": "Ron",
+            "LAST_NAME": "Mattys",
+            "CITY": "Bayaguana"
+          }
+        },
+        {
+          "node": {
+            "FIRST_NAME": "Althea",
+            "LAST_NAME": "Featherstone",
+            "CITY": "Calatrava"
+          }
+        },
+        {
+          "node": {
+            "FIRST_NAME": "Ivett",
+            "LAST_NAME": "Casemore",
+            "CITY": "Campina Grande"
+          }
+        }
+      ],
+      "pageInfo": {
+        "endCursor": "eyJjIjoiTDpRdWVyeTplbXBsb3llZXMiLCJvIjo0fQ==",
+        "hasNextPage": true
+      }
+    }
+  }
+}
+```
+
+Note that `pageInfo` contains information that is used to get the next page, `hasNextPage` indicates
+there is more data available. We now take the opaque `endCursor` and use it as the value for after to get the next five employees.
+```
+stepzen request 'query{employees(after:"eyJjIjoiTDpRdWVyeTplbXBsb3llZXMiLCJvIjo0fQ==",first:5){edges{node{FIRST_NAME LAST_NAME CITY}}pageInfo{endCursor hasNextPage}}}'
+```
+This has simply added the `after` argument, taken from `endCursor`:
+
+```graphql
+query {
+  employees(after:"eyJjIjoiTDpRdWVyeTplbXBsb3llZXMiLCJvIjo0fQ==" first: 5) {
+  # selection omitted
+  }
+}
+```
+As an aside, typically an application would handle pagination using variables, for example an operation like this,
```

Comment:
As an aside, an application would typically handle pagination using variables. For example, ...